Remove block dependency from block/utils - Closes #4012

michielmulders opened the pull request:
### What was the problem?
Passing block dependency to several files in the `block` folder creating circular dependency.
To solve this now, we pass the `blocksLogic` dependency.

However, as @nazarhussain indicated, we should create pure functions as util functions. Currently, they do processing and `I/O` with DB. 

### How did I solve it?
Pass `blocksLogic` dependency directly as argument.

### Review checklist

- [x] The PR resolves #4012
- [x] All new code is covered with unit tests
- [x] Relevant integration / functional tests are added
- [x] Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
- [x] Documentation has been added/updated

michielmulders commented:
To avoid circular dependency @shuse2

shuse2 commented:
I would prefer to clean up the code location than just injecting for that purpose

nazarhussain commented on framework/src/modules/chain/blocks/blocks.js:
```diff
@@ -293,6 +294,7 @@ class Blocks extends EventEmitter {
 				this.storage,
 				this.interfaceAdapters,
 				this.genesisBlock,
+				blocksLogic,
```
Agree that its the easiest way to solve. But I don't like that direction. We have to resolve and not pass any other module to utils. To me when util functions require instance of other modules then these are actually not standalone utils functions. We should rather move `loadLastBlock` to `blockLogic` where it relates more accurately.

michielmulders replied:
The loadLastBlock function is not the issue, it is that `dbRead` and `storageRead` require `blocksLogic`.

michielmulders commented:
@nazarhussain I moved the functions that were not pure util functions to `block.js`.
Also, implemented back the `sortTransactions` function which was initially causing the circular dep.

Currently, the dep tree looks like:
`block` -> `block_v1/2` -> `utils`

`blocks`
    -> `utils`
    -> `block`

`utils` -> No dependencies!

**Full dep list: `blocks` -> `block` -> `block_v1/2` -> `utils` END**

Before:
`utils` required `block` which caused the circular dependency.

shuse2 commented on framework/src/modules/chain/blocks/blocks.js:
```diff
@@ -17,6 +17,7 @@
 const EventEmitter = require('events');
 const { cloneDeep } = require('lodash');
 const blocksUtils = require('./utils');
+const Block = require('./block');
```
```suggestion
const blocksLogic = require('./block');
```
